version: '3.9'
services:

  django:
    build: .
    volumes:
     - static:/var/www/gutendex/static-root
    depends_on:
      postgres:
        condition: service_healthy
    hostname: django
    networks:
      - backend
      - frontend 

  postgres:
    image: postgres:14
    environment:
      POSTGRES_PASSWORD: 1234 
      POSTGRES_USER: gutendex
      POSTGRES_DB: gutendex
    volumes:
      - ./init.sql:/docker-entrypoint-initdb.d/init.sql
      - database:/var/lib/postgresql/data
    hostname: postgres
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready", "-q", "-d", "gutendex", "-U", "gutendex", '-p1234' ]
      interval: 10s
      timeout: 5s
      retries: 5
    networks:
      - backend

  nginx:
    image: nginx:1.23-alpine
    volumes:
      - ./nginx/nginx.conf:/etc/nginx/nginx.conf:ro 
      - ./project/static:/usr/share/nginx/html/static
      - ./project/gutendex/templates:/usr/share/nginx/html/
    networks:
      - frontend
    ports:
      - "80:80"
    depends_on:
      - django
      - postgres


volumes:
  database:
  static:

networks:
  backend:
  frontend:

